deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+        else:
+            raise Exception("Requested object_handling ({0}) not currently supported.".format(object_handling))
+        # Context
+        if include_context and self.context:
+            text_list.append(self.context)
+        # Currently, source, time and location aren't being used, but this can be changed in the future.
+
+        return text_list
+
+
+    @classmethod
+    def from_string(cls, input_string: str):
+        tuple_fields = input_string.lower().split("|")
+        num_fields = len(tuple_fields)
+        assert num_fields >= 2, "num_fields not >= 2" + input_string
+        subject = tuple_fields[0]
+        verb = tuple_fields[1]
+        objects = []
+        context, time, location, source = None, None, None, None
+        for element in tuple_fields[2:]:
+            if element.startswith("context:"):
+                context = element.split("context:")[1]
+            elif element.startswith("L:"):
+                location = element.split("L:")[1]
+            elif element.startswith("T:"):
+                time = element.split("T:")[1]
+            elif element.startswith("source:"):
+                source = element.split("source:")[1]
+            else:
+                objects.append(element)
+        return TextTuple(subject, verb, objects, context, time, location, source)
+
+
+class MCTuplesWithBackgroundInstance(TextInstance):
+    """
+    A MCTuplesWithBackgroundInstance is a kind of TextInstance that has text in multiple slots (stored
+    as a TextTuple), for both the answer choices and background information.
+    This can be used to store SVO triples.
+
+    Parameters
+    ----------
+    - answer_tuples: List[List[TextTuple]]
+        This is a list of length = num answer options.  Each answer has a list of TextTuples (made
+        from the question+answer_candidate tuples).
+
+    - background_tuples: List[TextTuple]
+        This is a list of background TextTuples (currently used for all answer candidates).
+
+    - object_handling: str, default='collapse'
+        This tells the Instance how to handle the situation where there are multiple objects.
+        Options include:
+            - "collapse" : collapse all of the objects into one object
+            - "first" : use only the first object (ignore the rest)
+
+    - use_context: bool, default=True
+        Lets the instance know if you want to retain a slot for information marked as context.  If False,
+        the context information is ignored.
+
+    - label: int, default=None
+        The class label (i.e. the index of the correct multiple choice answer) -- corresponds to the
+        indices in self.answer_tuples.
+
+    - index: int, default=None
+        The index of the question.
+
+    """
+    def __init__(self, answer_tuples: List[List[TextTuple]], background_tuples: List[TextTuple],
+                 object_handling: str="collapse", use_context: bool=True, label: int=None, index: int=None):
```

Comment:
Notice [this line](https://github.com/allenai/deep_qa/blob/aa1c81296866001d465287d98ff675c4ef9b3ea9/deep_qa/data/instances/instance.py#L77), where the tokenizer is used in the instance object.  There is no definition of `self.tokenizer` in the `self.__init__` method - `tokenizer` is a _class_ variable, not an _instance_ variable.  It's basically the same thing as `static` in java.  So, you can make these parameters _class_ variables, with reasonable defaults, and change them in the model class if you need to, by setting `TupleInferenceInstance.object_handling`, etc.  This sets it for every instance of the class, which is what you want (assuming you only ever want one behavior for a particular run of your program, which I think is a reasonable assumption).